File: src/my_py_pkg/my_py_pkg/hardware_status_publisher.py
import rclpy
from rclpy.node import Node

from my_robot_interfaces.msg import HardwareStatus

class HardwareStatusPublisherNode(Node):
    def __init__(self):
        super().__init__("hardware_status_publisher")
        self.number = 5
        self.publisher = self.create_publisher(HardwareStatus, "hardware_status", 10)
        self.timer = self.create_timer(1.0, self.publish_status)
        self.get_logger().info("Hardware Status Publisher started")
    
    def publish_status(self):
        msg = HardwareStatus(temperature=50.0, are_motors_ready=True, debug_message="OK")
        self.get_logger().info(f"sending {msg}")
        self.publisher.publish(msg)


def main(args=None):
    rclpy.init(args=args)
    rclpy.spin(HardwareStatusPublisherNode())
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/hardware_status_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "my_robot_interfaces/msg/hardware_status.hpp"


class HardwareStatusPublisherNode : public rclcpp::Node {
    public:
        HardwareStatusPublisherNode() : rclcpp::Node("hardware_status_publisher") {
            this->publisher = this->create_publisher<my_robot_interfaces::msg::HardwareStatus>("hardware_status", 10);
            this->_timer = this->create_wall_timer(std::chrono::seconds(1),
                                                   std::bind(&HardwareStatusPublisherNode::publishHardwareStatus, this));

            RCLCPP_INFO(this->get_logger(), "Hardware Status Publisher has started");
        }
    
    private:

        void publishHardwareStatus() {
            auto message = my_robot_interfaces::msg::HardwareStatus();
            message.temperature = 20.3;
            message.are_motors_ready = true;
            message.debug_message = "status ok";
            RCLCPP_INFO(this->get_logger(), "Sending status, temperature %f", message.temperature);
            this->publisher->publish(message);
        }

        rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr publisher;
        rclcpp::TimerBase::SharedPtr _timer;
};


int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<HardwareStatusPublisherNode>());    
    rclcpp::shutdown();
    return 0;
}